libparsec/crates/client/tests/unit/certif/validate_child_manifest.rs

```diff
@@ -196,6 +198,165 @@ async fn corrupted(#[values("dummy", "parent_pointing_on_itself")] kind: &str, e
         },
     );
 
+    let err = ops
+        .validate_child_manifest(
+            env.get_last_realm_certificate_timestamp(realm_id),
+            env.get_last_common_certificate_timestamp(),
+            realm_id,
+            realm_key_index,
+            child_id,
+            &alice.device_id,
+            0,
+            now,
+            &encrypted,
+        )
+        .await
+        .unwrap_err();
+
+    match kind {
```

Comment:
Small comment on style: in the tests I try to have a single `match kind` statement so that it can always have the same pattern:
- every case is matched
- unknown guard
- every expected value is generated there and returned from the match (so here we would have an `let expected_data_error: DataError = match kind { ... };` 

My main fear is to open the door to non exhaustive match (typically when somewhere in the code we need to do something for kind A and B but not for C and D, then we start have `match kind { ... }` without the unknown guard, which can then be copied from one test to another by new dev...)